docs/using-pre-commit.md

```diff
@@ -27,35 +27,41 @@ Note that these hooks do not update the `.ipynb` notebook when you pull. Make su
 
 ## Using Jupytext with the pre-commit package manager
 
-Using Jupytext with the [pre-commit package manager](https://pre-commit.com/) is another option. You could add the following to your `.pre-commit-config.yaml` file:
-```
+Using Jupytext with the [pre-commit package manager](https://pre-commit.com/) is another option. You could add the following to your `.pre-commit-config.yaml` file to convert all staged notebooks to python scripts in `py:percent` format (the default):
+
+```yaml
 repos:
--   repo: local
+-   repo: https://github.com/mwouts/jupytext
+    rev: master
     hooks:
     - id: jupytext
-      name: jupytext
-      entry: jupytext --to md
-      files: .ipynb
-      language: python
 ```
 
-Here is another `.pre-commit-config.yaml` example that uses the --pre-commit mode of Jupytext to convert all `.ipynb` notebooks to `py:light` representation and unstage the `.ipynb` files before committing.
-```
+You can also provide arguments to Jupytext in pre-commit, for example to produce several kinds of output files:
+
+```yaml
 repos:
-  -
-    repo: local
+-   repo: https://github.com/mwouts/jupytext
+    rev: master
     hooks:
-      -
-        id: jupytext
-        name: jupytext
-        entry: jupytext --from ipynb --to py:light --pre-commit
-        pass_filenames: false
-        language: python
-      -
-        id: unstage-ipynb
-        name: unstage-ipynb
-        entry: git reset HEAD **/*.ipynb
-        pass_filenames: false
-        language: system
+    - id: jupytext
+      args: [--to, py:light]
+    - id: jupytext
+      args: [--to, markdown]
+```
+
+If you are combining Jupytext with other pre-commit hooks, you must ensure that all hooks will pass on any files you generate. For example, if you have a hook for using `black` to format all your python code, then you should use Jupytext's `--pipe` option to also format newly generated Python scripts before writing them:
 
+```yaml
+repos:
+-   repo: https://github.com/mwouts/jupytext
+    rev: master
+    hooks:
+    - id: jupytext
+      args: [--to, py:percent, --pipe, black]
+-   repo: https://github.com/psf/black
```

Comment:
This isn't necessary or dieal and could conflict with otehr black hooks. It should just be added as an `additional_dependency` but the end user if needed at all.

Reply:
It's just an informational example

Reply:
Oh this is abusing the virtualenv of the pre-commit hook (it should only require the bear minimum requirements and other hooks if they absolutely build upon one another). It should really be installed using additional_depenencies if needed by the end user (see the blog post test I adjusted.

Reply:
Added black in additional_depenencies

Reply:
Yes I agree with `additional_dependencies` (I think there is a typo in `additional_depenencies`).

Also you say _you must ensure that all hooks will pass (...)_. 
I understand that we are obliged to do so when we use `--to`, but maybe if we use `--sync` it's just "more convenient" as we need to run the hook just once, not twice. Is that correct?

Reply:
@JohnPaton , what would you think of removing this example? I was trying to add a test for it, but we seem to have tricky issues - maybe we add that in another PR?
```
@requires_pre_commit
def test_pre_commit_hook_sync_black(tmpdir):
    # get the path and revision of this repo, to use with pre-commit
    repo_root = str(Path(__file__).parent.parent.resolve())
    repo_rev = system("git", "rev-parse", "HEAD", cwd=repo_root).strip()

    git = git_in_tmpdir(tmpdir)

    # set up the tmpdir repo with pre-commit
    pre_commit_config_yaml = dedent(
        f"""
        repos:
        - repo: {repo_root}
          rev: {repo_rev}
          hooks:
          - id: jupytext
            args: [--sync, --pipe, black]
            additional_dependencies:
              - black==19.10b0 # Matches hook
        
        - repo: https://github.com/psf/black
          rev: 19.10b0
          hooks:
          - id: black
            language_version: python3
        """
    )
    tmpdir.join(".pre-commit-config.yaml").write(pre_commit_config_yaml)
    git("add", ".pre-commit-config.yaml")
    with tmpdir.as_cwd():
        pre_commit(["install", "--install-hooks"])

    # write test notebook and output file
    nb = new_notebook(
        cells=[new_code_cell("1+1")],
        metadata={
            "jupytext": {"formats": "ipynb,py:percent", "main_language": "python"}
        },
    )
    nb_file = tmpdir.join("test.ipynb")
    py_file = tmpdir.join("test.py")
    write(nb, str(nb_file))

    git("add", ".")
    # First attempt to commit fails with message
    # "Output file test.py is not tracked in the git index"
    with pytest.raises(SystemExit):
        git("commit", "-m", "fails")

    git("commit", "-m", "succeeds")

    assert "test.ipynb" in git("ls-files")
    assert "test.py" in git("ls-files")
    # py_file should have been reformated by black
    assert "\n1 + 1\n" in py_file.read_text()
```